### ISSUE TITLE
Parse request body

### ISSUE THREAD

**Issue by clue** [reactphp/http#105]
We should provide an *easier* way to access the parsed request body, i.e. `$_POST` values from HTML forms, file uploads etc.

This depends on detecting actual message lengths first, see #104.

I'm filing this mostly for the reference, there's been some discussion to wait for PSR-7 message support (#28) first.

**Comment by clue** [comment]
For the reference: Parts of this had been implemented on the master branch but had to be reverted due to licensing and blocking issues, see https://github.com/reactphp/http/pull/59 for more details.

Some projects have started to build on top of this, so we're currently providing a dev branch for this: https://github.com/reactphp/http/tree/streaming-multipart. See also #41 for more details. Note that this is development preview only and we can only provide limited support for this.

In other words: If you run into issues with this branch, please refrain from opening new issues and comment on the respective PRs instead.

**Comment by clue** [comment]
> It's very likely the future version will look somewhat different than this development preview […]

Rest assured, we're working on hard on providing a stable base for this soon. Bear with us and we'll keep this ticket updated :+1:

**Comment by andig** [comment]
+1 for keeping the ability to access plain body. Otherwise it becomes horribly cumbersome to interface e.g. with symfony/httpkernel

**Comment by WyriHaximus** [comment]
@andig without any doubt we will allow users to have access to the raw body
